File: python/ServerTerminator.py
from Server import Server
from Status import Status
from DoublyLinkedList import DoublyLinkedList

class ServerTerminator:
    def __init__(self, queue: DoublyLinkedList[Server]):
        self.__queue = queue
    
    def terminateNextServer(self):
        server = self.__queue.dequeue().set_status(Status.TERMINATING)

File: js/ServerTerminator.js
const Status = require("./Status");

class ServerTerminator{
    #queue;
    
    constructor(queue){
        this.#queue = queue;
    }
    
    terminateNextServer(){
        this.#queue.dequeue().setStatus(Status.TERMINATING);
    }
}

module.exports = ServerTerminator;
